### ISSUE TITLE
The future of GD WAD Loader?

### ISSUE THREAD

**Issue by HeadClot** [Chaosus/GDWADLevelLoader#1]
Hey @Chaosus - What is the future of GDWADLevelloader? I noticed that there has not been a commit in several months.

I am super interested in this being developed further. :)

If you need any advice on what to tackle next as well as some code to reference let me know.

**Comment by Chaosus** [comment]
Hi. There was computer problems on my side(my hard disk with projects was crashed), gladly I restored it(with motivation to continue), I still want to update this project. I take a look on this today..

**Comment by Chaosus** [comment]
If you want to help and contribute you can look at http://www.gamers.org/dhs/helpdocs/dmsp1666.html
This is the full WAD format description. You can took the ideas from it.

**Comment by jminor** [comment]
You may also find this wad2obj.py source code helpful. It can extract full 3D mesh and textures from a WAD. https://github.com/jminor/omgifol/blob/master/demo/wad2obj.py

**Comment by Chaosus** [comment]
Thanks ! I will take a look on it

**Comment by HeadClot** [comment]
Hey just thought I would chime in again. 

There are a few projects that have working Doom Level Importers. 

https://github.com/jbaicoianu/wadjs - WadJS is written in Javascript and Uses the Three.js library. Might be useful as a reference. 

There is also Wad2Unity. https://bitbucket.org/freegodsoul/wad2unity This would be also useful as a reference. Written In C# though

**Comment by Chaosus** [comment]
Wad2Unity seems solid and I like C#, need time to research..

**Comment by Chaosus** [comment]
Damn, the wad2unity contains a lot of code for triangulation and other hard geometry stuff.. I doesnt sure if I could make it alone, need to take a pause. Anyway, the code is compiles and loading wad correctly.

**Comment by HeadClot** [comment]
@Chaosus - I know that Wad2Unity uses poly2Tri which can be found here. https://github.com/greenm01/poly2tri

**Comment by Chaosus** [comment]
@HeadClot  Yeah, there is pretty much code inside, need time to rest and research

**Comment by ghost** [comment]
@HeadClot @Chaosus If you need additional code reference for working with Doom formats, the source code for GZDoom Builder Bugfix is C# and fairly well organized and easy to get hold of. You may consider looking at it as well.

**Comment by HeadClot** [comment]
Just came across two things which may be super helpful for importing doom levels in Godot.

https://github.com/jmickle66666666/DoomUnity - This is a partially working implementation from my understanding.

https://medium.com/@jmickle_/build-a-model-of-a-doom-level-7283addf009f  - Written by the same person and talks about how they imported doom levels into Unity.

**Comment by Chaosus** [comment]
Ok, I take a look on this, soon..

**Comment by Chaosus** [comment]
Interesting that Godot have a method for triangulation called Geometry.triangulate_polygon that could be helpful and maybe no..

**Comment by vitoralmeidasilva** [comment]
Hey @Chaosus , based on this project I took a shoot on trying the same with Wolfestein 3D (mainly a map renderer using the original game data reverse engineered; currently it loads the geometry but I'm having some issues with the textures and compression).
When I finish it, I intend to help with this project.

**Comment by Chaosus** [comment]
Ok, thanks ^^

**Comment by vitoralmeidasilva** [comment]
Well, I've got some progress (regarding the Wolfenstein 3D level loader): https://twitter.com/vitoralmeidasil/status/1197307886352916480

**Comment by Chaosus** [comment]
Ok, I'm waiting. If you can prepare the PR to enhance this project, I will gladly accept it and update the Godot asset version

**Comment by HeadClot** [comment]
Hey @vitoralmeidasilva got a few questions 
1. Do you plan on supporting doom level formats? such as UDMF for example?
2. We we be able load custom Wolf 3D Maps into Godot via this plugin? After you make your PR ofc.
3. What version of Godot are you using? I am guessing 3.1 or 3.2.

**Comment by vitoralmeidasilva** [comment]
Hello HeadClot.

1. The Wolfenstein 3D is a separate project (for the moment); I do intend to finish it first (almost there), share the code,  and them work on GD WAD Loader (as time allows) to do the same with WAD maps: Daggerfall Unity and this project was the inspiration that ignited the idea);
2. The plan was initially to have a kind of Node that can load Wolf3D maps inside a given project (but the scope is increasing with time hahaha);
3. I've been developing the project since version 3.0.6 (I'm now using 3.1 and intend to migrate to 3.2 as soon as possible).

The project is simple: a few scenes + typed GDScript files (no C#) + original Wolfenstein 3D data types (WL1, WL6 files).

**Comment by HeadClot** [comment]
@vitoralmeidasilva Let me know when you make a PR for this. I wait for it with eager anticipation :)
